tests/cmdline/commands/test_presto.py

```diff
@@ -0,0 +1,31 @@
+"""Tests for ``verdi presto``."""
+
+import pytest
+from aiida.cmdline.commands.cmd_presto import default_profile_name, verdi_presto
+from aiida.manage.configuration.config import Config
+
+
+@pytest.mark.parametrize(
+    'profile_names, expected',
+    (
+        ([], 'presto'),
+        (['main', 'sqlite'], 'presto'),
+        (['presto'], 'presto-1'),
+        (['presto', 'presto-5', 'presto-2'], 'presto-6'),
+        (['presto', 'main', 'presto-2', 'sqlite'], 'presto-3'),
+    ),
+)
+def test_default_profile_name(monkeypatch, profile_names, expected):
+    """Test the dynamic default profile function."""
+
+    def get_profile_names(self):
+        return profile_names
+
+    monkeypatch.setattr(Config, 'profile_names', property(get_profile_names))
+    assert default_profile_name() == expected
+
+
+def test_presto(run_cli_command):
+    """Test the ``verdi presto``."""
+    result = run_cli_command(verdi_presto, use_subprocess=False)
+    assert 'Created new profile `presto`.' in result.output
```

Comment:
This seems to be a very lenient test. Should we assert more stuff?

- that the profile actually exist and has been set as default.
- localhost computer has been setup
- maybe even try to run a workchain on it

In other words, a more end-to-end test would be good to verify that indeed everything is setup and one can use AiiDA straightaway.

Reply:
I updated the test to actually load the profile and check that the `localhost` exists and is configured, which also means that there is a default user. I don't think the test should actually try to run a process. These are unit tests and as you say that would be more an integration test. We could add this elsewhere, but I think this should be sufficient.

Reply:
Agreed, thanks!

One more thing if I can, please test that the rabbitmq auto detection works as expected.

Reply:
Done